shadows/framework/src/main/java/org/robolectric/shadows/ShadowLegacyAssetManager.java

```diff
@@ -235,29 +235,24 @@ private void convertAndFill(AttributeResource attribute, TypedValue outValue, Re
 
 
   public TypedResource getAttrTypeData(ResName resName) {
-    return resourceTable.getValue(resName, config);
+    return getResourceTable().getValue(resName, config);
   }
 
   @Implementation
   protected void __constructor__() {
-    resourceTable = RuntimeEnvironment.getAppResourceTable();
```

Comment:
Is there any reason to delay initialization of `resourceTable`? If not, what about storing `isSystem`, and changing this `resourceTable = *` to `resourceTable = isSystem ? RuntimeEnvironment.getSystemResourceTable() : RuntimeEnvinronment.getAppResourceTable()` only?

Reply:
AFAIK, it's necessary because of the 2nd constructor below that takes isSystem as a parm...

Reply:
Looks like the second constructor is called, and its calling-chain calls the first constructor, so that the `resourceTable` is rewritten to app resource table.

Reply:
At least, my tests tell me that my assumption is not correct.

Reply:
Current tests show that when initializing system `ShadowLegacyAssetManager`, the `RuntimeEnvironment.getSystemResourceTable()` is null.